CMakeLists.txt

```diff
@@ -1,19 +1,105 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.5.2 FATAL_ERROR)
+####################################
+# CMake options
+####################################
 
-### CMP0025   Compiler id for Apple Clang is now AppleClang.
-### CMP0042   MACOSX_RPATH is enabled by default.
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5.2 FATAL_ERROR)
 
-FOREACH (p
-        CMP0025 # CMake 3.0
-        CMP0042 # CMake 3.0
-        )
-    IF (POLICY ${p})
-        cmake_policy(SET ${p} NEW)
-    ENDIF ()
-endforeach ()
+# Set cmake policy by version: https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html
+if(${CMAKE_VERSION} VERSION_LESS 3.12)
+    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+else()
+    cmake_policy(VERSION 3.12)
+endif()
 
 enable_testing()
 
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+include(CMakeParseArguments)
+include(QCModulesUtils)
+
+####################################
+# Project
+####################################
+
+project(QualityControl
+  VERSION 0.6.2   # TODO update this automatically when there are new releases
+  DESCRIPTION "O2 Quality Control"
+  LANGUAGES CXX
+  )
+
+set(LIBRARY_OUTPUT_PATH "${CMAKE_BINARY_DIR}/lib")
+set(EXECUTABLE_OUTPUT_PATH "${CMAKE_BINARY_DIR}/bin")
+set(INCLUDE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/include")
+
+####################################
+# Compilation flags and build options
+####################################
+
+# Set the default build type to "RelWithDebInfo"
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "RelWithDebInfo"
+      CACHE
+      STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel Coverage."
+      FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
+# Build targets with install rpath on Mac to dramatically speed up installation
+# TODO reevaluate if this is needed
```

Comment:
I think it is, see https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling

Reply:
I really don't want to read this long page, so I will just trust you and remove the TODO.